pkg/scheduler/extender/extender.go

```diff
@@ -135,9 +137,41 @@ func (e *Extender) FilterHandler(w http.ResponseWriter, req *http.Request) {
 }
 
 // PrioritizeHandler assigns scores to the nodes
-// todo not implemented
 func (e *Extender) PrioritizeHandler(w http.ResponseWriter, req *http.Request) {
-	panic("implement me")
+	sessionUUID := uuid.New().String()
+	ll := e.logger.WithFields(logrus.Fields{
+		"sessionUUID": sessionUUID,
+		"method":      "PrioritizeHandler",
+	})
+	ll.Infof("Processing request: %v", req)
+
+	w.Header().Set("Content-Type", "application/json")
+	resp := json.NewEncoder(w)
+
+	var (
+		extenderArgs schedulerapi.ExtenderArgs
+	)
+
+	if err := json.NewDecoder(req.Body).Decode(&extenderArgs); err != nil {
+		ll.Errorf("Unable to decode request body: %v", err)
+		return
+	}
+
+	ll.Info("Scoring")
+
+	e.Lock()
```

Comment:
Do we really need to have a lock here?

Reply:
@mishoyama It's not 100% required here, but It will help with parallel requests, so ranking will be more precise.
Also, the lock will protect Kubernetes API from being spammed, If we do lots of parallel requests to this endpoint.